### DISCUSSION TITLE
Call python script from svg-editor.js

### DISCUSSION THREAD

**Discussion by michaelliesenberg** [SVG-Edit/svgedit#590] [Q&A]
Hi,

i want to call my python script in svg-editor.js file like this:
`const clickRun = function () {
    if (toolButtonClick('#tool_run'))
    {
        const spawn = require("child_process").spawn;
        const pythonProcess = spawn('python', "path/to/script.py");
        pythonProcess.stdout.on('data', (data) => {
          // Do something with the data returned from python script
        });
    }
  };`

But i get an error "Uncaught ReferenceError: require is not defined"

What do i need to do to make it work?

**Comment by jfhenon** [comment]
Have you already tried to find a solution outside svgedit? I'm not sure you can spawn a python process from a browser.

**Reply by michaelliesenberg** [reply]
I am still trying and found this [website](https://ourcodeworld.com/articles/read/286/how-to-execute-a-python-script-and-retrieve-output-data-and-errors-in-node-js)

Now i download required.js from [here](https://requirejs.org/docs/download.html) 
added to my svgedit project and in "svg-editor-es.html" i add this

    <script src="require.js"></script>

So now it can find the require function and once i press the button i call this
`require(["child_process"], function (child_process) {
        // child_process are now loaded
        });`

But i get this error 
"Failed to load resource: the server responded with a status of 404 (Not Found)"
"Uncaught Error: Script error for "child_process""

I went in to my project and did
`npm install child_process`

But i am still getting this error

Any idea?
